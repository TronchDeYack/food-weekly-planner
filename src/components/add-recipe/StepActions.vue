<template>
  <div class="mt-2">
    <v-btn class="continue-btn" color="primary" @click.native="nextStep">
      {{ $t('common.btn.continue') }}
    </v-btn>
    <v-btn class="cancel-btn" text @click.native="previousStep">
      {{ $t('common.btn.cancel') }}
    </v-btn>
  </div>
</template>

<script>
import { ADD_RECIPE_NEXT_STEP, ADD_RECIPE_PREVIOUS_STEP } from '@/store/mutations';

export default {
  name: 'add-recipe-step-actions',
  methods: {
    nextStep() {
      this.$store.commit(ADD_RECIPE_NEXT_STEP);
    },
    previousStep() {
      this.$store.commit(ADD_RECIPE_PREVIOUS_STEP);
    },
  },
};
</script>
