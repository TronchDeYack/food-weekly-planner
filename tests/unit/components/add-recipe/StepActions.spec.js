import StepActions from '@/components/add-recipe/StepActions.vue';

import { createComponent } from '../../../utils';

describe('StepActions', () => {
  let wrapper;

  const currentStep = 2;

  const propsData = {
    step: currentStep,
  };

  beforeEach(() => {
    wrapper = createComponent(StepActions, propsData);
  });

  describe('render', () => {
    it('Should render.', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('nextStep', () => {
      it('Should emit a "next-step" event with the next step value when clicking on continue button.', () => {
        const expectedContent = { step: currentStep + 1 };
        wrapper.find('.continue-btn').trigger('click');

        expect(wrapper.emitted()['next-step'][0]).toEqual([expectedContent]);
      });
    });

    describe('previousStep', () => {
      it('Should emit a "previous-step" event with the previous step value when clicking on cancel button.', () => {
        const expectedContent = { step: currentStep - 1 };
        wrapper.find('.cancel-btn').trigger('click');

        expect(wrapper.emitted()['previous-step'][0]).toEqual([expectedContent]);
      });
    });
  });
});
