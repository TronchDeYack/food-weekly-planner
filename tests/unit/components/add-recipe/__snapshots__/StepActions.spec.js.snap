// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepActions render Should render. 1`] = `
<div class="mt-2">
  <v-btn-stub color="primary" tag="button" activeclass="" type="button" class="continue-btn">
    common.btn.continue
  </v-btn-stub>
  <v-btn-stub tag="button" activeclass="" text="true" type="button" class="cancel-btn">
    common.btn.cancel
  </v-btn-stub>
</div>
`;
